File: awa/awa/sdk/utils/activity/resolve_template_activity.py
from datetime import timedelta
from typing import Any

from temporalio import workflow

from awa.sdk import constants


async def resolve_template_activity(
    template_str: str,
    variables: dict[str, Any] | None = None,
) -> str:
    """Resolve a template string by substituting variables.

    Args:
        template_str: The template string containing variable placeholders.
        variables: Optional dictionary of variables to substitute in the template.
            Defaults to None.

    Returns:
        str: The resolved template string with variables substituted.

    """
    return await workflow.execute_activity(
        constants.ACTIVITY_RESOLVE_TEMPLATE,
        args=[template_str, variables],
        task_queue=constants.AWA_DEFAULT_TASK_QUEUE,
        start_to_close_timeout=timedelta(seconds=constants.FILE_IO_TIMEOUT_SECONDS),
    )

File: awa/sdk_dist/csharp/Awa.Client/Utilities/Activity/ResolveTemplateActivity.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Awa.Client.Constants;

namespace Awa.Client.Utilities.Activity
{
    public static partial class ActivityUtilities
    {
        /// <summary>
        /// Resolve a template string by substituting variables.
        /// </summary>
        /// <param name="templateStr">The template string containing variable placeholders.</param>
        /// <param name="variables">Optional dictionary of variables to substitute in the template. Defaults to null.</param>
        /// <returns>The resolved template string with variables substituted.</returns>
        public static async Task<string?> ResolveTemplateActivity(
            string templateStr,
            Dictionary<string, object?>? variables = null)
        {
            return await Temporalio.Workflows.Workflow.ExecuteActivityAsync<string?>(
                AwaConstants.ActivityResolveTemplate,
                new object?[] { templateStr, variables },
                new()
                {
                    TaskQueue = AwaConstants.AwaDefaultTaskQueue,
                    StartToCloseTimeout = TimeSpan.FromSeconds(AwaConstants.FileIoTimeoutSeconds)
                });
        }
    }
}
